### ISSUE TITLE
Filter user's profile posts by tags

### ISSUE THREAD

**Issue by oct2pus** [pixelfed/pixelfed#562]
## What
In light of tumblr cracking down on nsfw art as well as greatly reducing their artists ability to monetize and thrive by removing any posts from searches and tags that have a link on or off site I think we should integrate deviantart-like gallery filter using tags (like tumblr), reasoning explained below.

For reference here is the deviant art 'gallery' view.

![image](https://user-images.githubusercontent.com/30452393/48733675-f50d8e80-ec08-11e8-9cd0-ed30a343523c.png)

if you click on a 'folder' to the left, it filters the overall view to just things put into that folder.

![image](https://user-images.githubusercontent.com/30452393/48733837-5a617f80-ec09-11e8-84b3-090084de3b8f.png)

I propose we add a panel to the left that would display tags that the user has used, and clicking on it would only display pictures that contain that tag.

![image](https://catgirl.science/media/ee77e78e-db57-44ef-b387-73df87427782/mockup.png)
#ProgrammerArt

if you click on a tag, you are only displayed content related to that tag. one could possibly click multiple tags to display only posts that contain all tags selected. Tags should be organized by frequency but users should be allowed to pin certain tags to the top (I don't really have a UI idea for this, possibly something in settings or a little pin emoji on tags when users view their own gallery).

Here is how I imagine it interacts with Post visibility in this way:
1. Public is public, it's both displayed on the timeline as well as galleries and tags.
2. Unlisted would have art only appear when you click on a tag that it contains.
3. Locked would not display the art in the gallery, regardless of tagged status.

## Why

1. Granular searching promotes exploring user content, which is great for creatives because it helps expose their work which can be hard on a network of communities that by design create content that is fleeting and in-the-moment. 
2. The Fediverse is composed of multiple communities running multiple backends like Mastodon and Pleroma, which means that a larger assortment of people will be could be browsing this work than simply sharing it on a site like DeviantArt, which only has users who explicitly set out to look at the creators deviantart page. Artists would often use sites like tumblr and twitter not because they're great platforms for displaying art (they're actually really bad at that), but because they have the most discoverability which allows them to potentially sustain their livelihood or just find an audience in their craft.
3. Tags not only allow easier discovery of desired content, but allow a greater ability to filter content one does not *want* to see through filtering, promoting the usage of tagging in the Fediverse to enhance individual user experiences.
4. This helps differentiate Pixelfed from Instagram, why should we simply be content to provide the exact same service but federated, we should be imagining new ways of communicating that value the users over corporations.
5. This provides functionality similar to what artists expect when they come in from other social networks. Sudden loss of functionality because you don't enjoy corporate control of your social media experience can reduce uptake and continued use.

tl;dr: this is a great way to allow gallery views while promoting a form of non-fleeting content in the Fediverse. It also helps differentiate Pixelfed from Instagram.
